### ISSUE TITLE
Prevent app from switching network automatically, require explicit user permission

### ISSUE THREAD

**Issue by kyranjamie** [leather-io/extension#1281]
Following on from #1280

**How it works now**
Connected apps can open transactions that transparently switch between a network _other_ than the one your wallet is currently set to, provided its in your list.

**Why this is a problem**
- **It's unexpected**: We provide no guidance/feedback that this switching happens (a user can only tell if they select the network switcher, or notice their balance is different)
- **An attack vector exists**: Consider you're in mainnet mode. A malicious "testnet" app guides you to switch to testnet. _"Great, I'm operating in a safe environment"_ you think. The app reassures you "_You're in testnet mode_". Now send me "testnet" STX, and the transaction window opens—the app switches it to a mainnet transaction and you don't notice the network mainnet label ⚠️ 

**Solution**
We implement the following behaviours:
1) The wallet can only operate in one network mode at any given time
2) The user **must** give their approval for an app to change network (regardless of whether you have it saved)

The flows are then:

![image](https://user-images.githubusercontent.com/1618764/121541196-5daaf980-ca07-11eb-9c71-dcae52231dc6.png)

See [EIP-2015](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-2015.md) best practices which notes similar function.

cc/ @jasperjansz @fbwoolf @andresgalante @aulneau

**Comment by jasperjansz** [comment]
I'm on board, big improvement. Should we also remove the ability to switch network in the tx signing screen? It will just lead to the contract not being found.

**Comment by aulneau** [comment]
big +1 for me. I think the only thing we should really allow is to switch user, but also warn them about that.

Similar to this is an app can be signed into one account, but the user could be on a different account. Do we want to show a warning for this too?

a thing I'd like to avoid is having multiple pop ups in our window

**Comment by fbwoolf** [comment]
I definitely agree this is critical to address and like the direction. 👍

**Comment by markmhendrickson** [comment]
> Connected apps can open transactions that transparently switch between a network other than the one your wallet is currently set to, provided its in your list.

Is this still true these or have we resolved this issue by now?

**Comment by kyranjamie** [comment]
True. This remains an issue, we've never built a change network confirmation screen.

**Comment by kyranjamie** [comment]
[Coinbase Wallet](https://blog.coinbase.com/making-web3-and-the-world-of-crypto-more-user-friendly-and-accessible-7ceda4b844d3) has added this same functionality/flow

> we have also added support for EIP-3085, which allows dapps to suggest a specific network to the user. When a dapp wants to suggest a network that differs from the user’s current network, Coinbase Wallet will ask the user to confirm the switch.

**Comment by markmhendrickson** [comment]
Reference UI from MetaMask:

<img width="374" alt="Screenshot 2023-06-15 at 15 07 05" src="https://github.com/hirosystems/wallet/assets/28991/933825c9-f2e8-4b96-8d9a-99c6407a05c1">
